src/main/GCNode.cpp

```diff
@@ -95,12 +106,27 @@ static bool test_allocated_bit(void* allocation) {
     return GC_is_marked(allocation);
 }
 
+// Computes the bin to update in an allocation frequency table.
+static int get_bin_number(size_t bytes) {
+    size_t bin;
+    if (bytes >= 32 * 8) {
+        bin = 31;
+    } else if (bytes >= 8) {
```

Comment:
Use 
   assert(bytes >= 8)
instead.  You're implicitly assuming that it's true here anyway (bin would be left uninitialized otherwise).